### PR TITLE
Replace approxEqual with isClose

### PR DESCRIPTION
**Pull request by berni44**
This is a copy of #7340. I rebased everything on master, removed conflicts and fixed one new place where approxEqual sneaked in in the last year.

### CODE REVIEW

**Comment by ibuclaw** [comment]
If I haven't already said on the previous PR, it would be good to have testing done on ARM64 at the very least to confirm that the new limits also suit 128-bit real platforms, and possibly a 64-bit real target as well.  `approxEqual` has already been widely tested, so the fewer number of times we have to fiddle with these unit-tests the better.

**Comment by berni44** [comment]
> If I haven't already said on the previous PR, it would be good to have testing done on ARM64 at the very least to confirm that the new limits also suit 128-bit real platforms, and possibly a 64-bit real target as well.

Any ideas, how we could do that? (And yes, I support this, but I have no means to do so.)

> `approxEqual` has already been widely tested, so the fewer number of times we have to fiddle with these unit-tests the better.

I disagree here. `approxEqual` has a large flaw, because it pretends to guard against mistakes, but doesn't in all cases. It performs (with defaults) only a very weak check. In case, a unittest fails with `isClose`, we should be happy, because this high likely points to a bug, that was left undetected before.

**Comment by ibuclaw** [comment]
> > `approxEqual` has already been widely tested, so the fewer number of times we have to fiddle with these unit-tests the better.
> 
> I disagree here. `approxEqual` has a large flaw, because it pretends to guard against mistakes, but doesn't in all cases. It performs (with defaults) only a very weak check. In case, a unittest fails with `isClose`, we should be happy, because this high likely points to a bug, that was left undetected before.

Let me clarify.  All unit-tests pass on Linux/arm32, Linux/arm64, Solaris/sparc, FreeBSD/i386, and a number of others with the current `approxEqual` checks.

**Comment by berni44** [comment]
> > > `approxEqual` has already been widely tested, so the fewer number of times we have to fiddle with these unit-tests the better.
> > 
> > 
> > I disagree here. `approxEqual` has a large flaw, because it pretends to guard against mistakes, but doesn't in all cases. It performs (with defaults) only a very weak check. In case, a unittest fails with `isClose`, we should be happy, because this high likely points to a bug, that was left undetected before.
> 
> Let me clarify. All unit-tests pass on Linux/arm32, Linux/arm64, Solaris/sparc, FreeBSD/i386, and a number of others with the current `approxEqual` checks.

Yes, that's clear. My point is, that the strength of unittests using `approxEqual` is sometimes very weak. An even weaker unittest `assert(true);` would also pass on all these platforms. But of course it won't check much.

File: std/algorithm/comparison.d [hunk 1]
```diff
@@ -998,8 +998,8 @@ template equal(alias pred = "a == b")
     assert(equal(a[], b[]));
 
     // predicated: ensure that two vectors are approximately equal
-    double[4] c = [ 1.005, 2, 4, 3];
-    assert(equal!approxEqual(b[], c[]));
+    double[4] c = [ 1.0000000005, 2, 4, 3];
```
Comment (RazvanN7) [inline std/algorithm/comparison.d:1001]: Why is this test changed?

File: std/parallelism.d [hunk 2]
```diff
@@ -4565,8 +4565,12 @@ version (StdUnittest)
 // tons of stuff and should not be run every time make unittest is run.
 version (parallelismStressTest)
 {
-    @safe unittest
+    @system unittest
```
Comment (RazvanN7) [inline std/parallelism.d:4568]: why is this unittest becoming `@system` ?

File: std/parallelism.d [hunk 3]
```diff
@@ -4565,8 +4565,12 @@ version (StdUnittest)
...
     {
+        import std.stdio : stderr, writeln, readln;
+        import std.range : iota;
+        import std.algorithm.iteration : filter, reduce;
```
Comment (RazvanN7) [inline std/parallelism.d:4572]: Why is this PR adding these imports? I see that these functions were used before adding these imports here.

File: std/parallelism.d [hunk 4]
```diff
@@ -4648,8 +4652,14 @@ version (parallelismStressTest)
 
     // These unittests are intended more for actual testing and not so much
     // as examples.
-    @safe unittest
+    @system unittest
```
Comment (RazvanN7) [inline std/parallelism.d:4655]: Same comments as for the previous unittest.

File: std/parallelism.d [hunk 5]
```diff
@@ -4776,6 +4787,7 @@ version (parallelismStressTest)
                     uselessTask.workForce();
                 }
             }
+             +/
```
Comment (RazvanN7) [inline std/parallelism.d:4790]: unrelated change. please undo.

**Review by RazvanN7 — APPROVED** [review]
Looks good. See my comments.

**Comment by RazvanN7** [comment]
@ibuclaw Our testing pipeline seems fine. How can we move forward here? How can we test this in a satisfactory way?
